blog/models.py

```diff
@@ -0,0 +1,33 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils.text import slugify
+
+from core.models import BaseModel
+
+
+class Post(BaseModel):
+    title = models.CharField(max_length=200)
+    slug = models.SlugField(
+        blank=True,
+        max_length=250,
+        null=True,
+        unique_for_date="published_date",
+    )
+    content = models.TextField()
+    author = models.ForeignKey(User, blank=True, on_delete=models.CASCADE, null=True)
+    published_date = models.DateTimeField(blank=True, null=True)
+
+    class Meta:
+        ordering = ["-published_date"]
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
+        # If the post is being published for the first time, generate a slug
+        if self.published_date and not self.slug:
+            # Format the published_date to include only the date portion
+            date_str = self.published_date.strftime("%Y-%m-%d")
+            self.slug = slugify(f"{date_str}-{self.title}")
```

Comment:
Would we want to have the date in the slug, or in the path of the url? Something like this

`2024/02/24/another-test`

vs this

`2024-02-24-another-test`

I'm not sure if there is an SEO hit (or if we care if there is)

Reply:
Good question. I have been eyeballing https://obsidian.md/changelog/ and leaning towards `2024-02-24-another-test` for inspiration. I have a List + Detail view on the backend to keep things easier.